Kadai

rkieee opened the pull request:
最新です。

rkieee committed:
commitしたお

rkieee committed:
kadai

rkieee committed:
wwwww

rkieee committed:
Delete index.html.html

rkieee committed:
Delete app.js.js

rkieee committed:
Update app.js.js

manaberuit commented on fizzbuzz/index.html.html:
```diff
@@ -0,0 +1,24 @@
+    <title>Fizz Buzz</title>
+</head>
+<body>
+    <header><h1>FizzBuzz問題</h1></header>
```
１行１要素として記述すると良いですよ。
業務ではそのようにコーディング規約を定めていることも多いです。

```
<header>
  <h1>FizzBuzz問題</h1>
</header>
```

manaberuit commented on fizzbuzz/index.html.html:
```diff
@@ -0,0 +1,24 @@
+        <br>
+        <label><span>BuzzNum:</span><input type="text" id="buzzValue" placeholder="整数を入力してください"></label>
+        <br>
+        <button type="submit" id="exe" onclick="buzzIt()">実行</button>
```
onclickは使用しない方が良いです。
業務では基本使用しないです。

というのも、htmlとjsはなるべく分割したいからです。
onclickを使用するとhtmlにjsの関数名を入れなければなりません。
これを避けるために、addEventListenerを使用した方が良いです。
https://developer.mozilla.org/ja/docs/Web/API/EventTarget/addEventListener

manaberuit commented on fizzbuzz/app.js.js:
```diff
@@ -0,0 +1,30 @@
+    let returnValue = '';
+    let condition = /^[0-9]+$/;
+    if(condition.test(value1) && condition.test(value2)){
+        for(i = 1;i <= 100;i++){
```
`;`の後には空白を入れましょう
```
for(i = 1; i <= 100; i++){
```

manaberuit commented on fizzbuzz/app.js.js:
```diff
@@ -0,0 +1,30 @@
+    let condition = /^[0-9]+$/;
+    if(condition.test(value1) && condition.test(value2)){
+        for(i = 1;i <= 100;i++){
+            if(i%value1 == 0 && i%value2 == 0){
```
`%`のような演算子の前後にも空白を入れることが一般的ですね。
```
if(i % value1 == 0 && i % value2 == 0){
```

manaberuit replied:
`==`より`===`を使用することが推奨されます。
`==`では勝手に型変換されてしまうため、バグの発生の原因になりやすいです。
型変換が必要な場面であっても`===`を使用して明示的に型変換しましょう。

manaberuit reviewed (commented):
いくつかコメントしましたので、ご確認ください。

manaberuit commented:
>最終的に正規表現ではできたのですがなぜtypeofを使ってできなかったのかがいまだにわかりません。

今回、value1やvalue2には文字列が格納されています。
フォームに数値を入力されても、`value`で取得すると、文字列として扱われます。

ですので、今回はtypeofを使用できなかったということですね。

rkieee committed:
Deleting onclick

ｈｔｍｌとｊｓは分割したほうがいいため、
ｈｔｍｌにｊｓの関数名を入れることは避ける。

rkieee committed:
Fixed js code

等価演算子(==) 暗黙の型変換をして比較する
厳密等価演算子(===) オペランド同士を型変換することなく、型や値を厳密に比較する
https://qiita.com/BINO/items/e0551c79d247d8f6ff28

rkieee commented:
修正しました！
「フォームに数値を入力されても、valueで取得すると、文字列として扱われる。」
これで数日頭抱えていたので絶対もう忘れないと思います。

manaberuit commented on fizzbuzz/app.js.js:
```diff
@@ -0,0 +1,31 @@
+//Parameter driven
+function FizzBuzz(value1, value2){
+    let returnValue = '';
+    let condition = /^[0-9]+$/;
```
この変数は再代入しないので、constで良い気がしますね。
なるべくconstを使用するように心がけると良いですよ^^
https://qiita.com/wannabe/items/b2a0d63fc786eab13c48

manaberuit reviewed (approved):
1点だけコメントしました。
大した指摘ではないので、修正後の再レビューはなくてOKです。

また、現在は私のリポジトリに対してプルリクエストして頂いておりますが、今後はご自身のリポジトリのmasterブランチに対してプルリクエストしてください^^
